src/plugins/safeSnap/utils/index.ts

```diff
@@ -20,7 +20,7 @@ export const mustBeEthereumContractAddress = memoize(
     const contractCode = await provider.getCode(address);
 
     return (
-      contractCode && contractCode.replace('0x', '').replace(/0/g, '') !== ''
+      contractCode && contractCode.replace(/^0x/, '').replace(/0/g, '') !== ''
```

Comment:
this is an unrelated change to fix the bug, but it seems like this could also potentially cause problems